### PR TITLE
Use productcode for stock card

### PR DESCRIPTION
**Pull request by danniyu**

### CODE REVIEW

File: src/main/java/org/openlmis/stockmanagement/repository/mapper/StockCardMapper.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
   @Select("SELECT *" +
       " FROM stock_cards" +
       " WHERE facilityid = #{facilityId}" +
-      "   AND productid = #{productId}")
+      "   AND productid = (SELECT id FROM products WHERE code = #{productCode})")
```
Comment (chunky56) [inline src/main/java/org/openlmis/stockmanagement/repository/mapper/StockCardMapper.java:17]: You may want to redo this SQL query to not use a subquery.

File: src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java [hunk 2]
```diff
@@ -94,7 +94,7 @@
 
     //TODO: Determine what the permissions associated with @PreAuthorize should be. (MANAGE_PROGRAM_PRODUCT, below, is just a placeholder).
 
-    @RequestMapping(value = "facilities/{facilityId}/products/{productId}/stockCard", method = GET, headers = ACCEPT_JSON)
+    @RequestMapping(value = "facilities/{facilityId}/products/{productCode}/stockCard", method = GET, headers = ACCEPT_JSON)
     @ApiOperation(value = "Get information about the stock card for the specified facility and product.",
```
Comment (chunky56) [inline src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java:98]: Please change the API documentation to reflect the change to the URL. (The documentation in the @ApiOperation section)

**Comment by chunky56** [comment]
Hey Danni,
I have reviewed the code with a couple of comments. Once you make the change at least to the documentation, you can go ahead and merge the pull request.

File: src/main/java/org/openlmis/stockmanagement/repository/mapper/StockCardMapper.java (hunk 1 above)
Comment (danniyu) [inline src/main/java/org/openlmis/stockmanagement/repository/mapper/StockCardMapper.java:17]: What's your suggestion?

Comment (chunky56) [inline src/main/java/org/openlmis/stockmanagement/repository/mapper/StockCardMapper.java:17]: I would have the query do a join with the products table using productId, and then have products.code in the where clause.

**Pull request merged**